src/cmdargs.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use clap::Parser;
+use std::path::PathBuf;
+
+// Command-line options of azure-init binary.
+// Each option is coupled with an environment variable of the option name,
+// e.g. run either with "azure-init --hostnamectl=/usr/sbin/hostnamectl" or
+// "HOSTNAMECTL=/usr/sbin/hostnamectl azure-init"
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct CmdArgs {
+    /// Path to hostnamectl
+    #[clap(long, env)]
+    hostnamectl: Option<PathBuf>,
```

Comment:
Is there a reason to not have the requirement that `PATH` is set up reasonably? [Command](https://doc.rust-lang.org/std/process/struct.Command.html#method.new) will use `PATH` to resolve the program if it's not an absolute path.